src/main/scala/lms/core/stub.scala

```diff
@@ -803,37 +808,43 @@ object PrimitiveTypeLess {
 
   class NUM(override val x: Backend.Exp) extends TOP(x) {
     def +(y: NUM)(implicit pos: SourceContext): NUM = {
-      assert(t == y.t)
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t} in NUM + method")
```

Comment:
```suggestion
      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t}")
```
When the assert fails, the call stack (as well as the last method) will be printed on the screen.